tests/components/yeelight/test_config_flow.py

```diff
@@ -230,32 +234,55 @@ async def test_options(hass: HomeAssistant):
         await hass.async_block_till_done()
 
     config = {
+        CONF_NAME: NAME,
         CONF_MODEL: "",
         CONF_TRANSITION: DEFAULT_TRANSITION,
         CONF_MODE_MUSIC: DEFAULT_MODE_MUSIC,
         CONF_SAVE_ON_CHANGE: DEFAULT_SAVE_ON_CHANGE,
         CONF_NIGHTLIGHT_SWITCH: DEFAULT_NIGHTLIGHT_SWITCH,
     }
-    assert config_entry.options == {
-        CONF_NAME: "",
-        **config,
-    }
+    assert config_entry.options == config
     assert hass.states.get(f"light.{NAME}_nightlight") is None
 
     result = await hass.config_entries.options.async_init(config_entry.entry_id)
     assert result["type"] == "form"
     assert result["step_id"] == "init"
 
     config[CONF_NIGHTLIGHT_SWITCH] = True
+    user_input = {**config}
+    user_input.pop(CONF_NAME)
     with patch(f"{MODULE}.Bulb", return_value=mocked_bulb):
         result2 = await hass.config_entries.options.async_configure(
-            result["flow_id"], config
+            result["flow_id"], user_input
         )
         await hass.async_block_till_done()
     assert result2["type"] == "create_entry"
-    assert result2["data"] == {
-        CONF_NAME: "",
-        **config,
-    }
+    assert result2["data"] == config
     assert result2["data"] == config_entry.options
     assert hass.states.get(f"light.{NAME}_nightlight") is not None
+
+
+async def test_manual_no_capabilities(hass: HomeAssistant):
+    """Test manually setup without successful get_capabilities."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    mocked_bulb = _mocked_bulb()
+    type(mocked_bulb).get_capabilities = MagicMock(return_value=None)
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb), patch(
+        f"{MODULE}.async_setup", return_value=True
+    ), patch(
+        f"{MODULE}.async_setup_entry",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_HOST: IP_ADDRESS}
+        )
+    type(mocked_bulb).get_capabilities.assert_called_once()
+    type(mocked_bulb).get_properties.assert_called_once()
+    assert result["type"] == "create_entry"
+    assert result["data"] == {CONF_HOST: IP_ADDRESS}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)